### DISCUSSION TITLE
Asking help for direction of PR #34715

### DISCUSSION THREAD

**Discussion by starone99** [apache/airflow#37045] [Q&A] [answered]
I have made a PR to fix what I perceived as a bug due to a fixed dag run timeout in the ExternalTaskSensor. However, I received various feedback, and was advised to add a new parameter while maintaining the existing format for backward compatibility.

In my opinion, adding a new parameter is not ideal, and I'm also struggling to come up with a suitable name for it. I feel that no name would be appropriate since there's already a 'timeout' and an 'execution_timeout'.

[Additionally, I've heard that submitting an AIP (Airflow Improvement Proposal) is another way to make fundamental changes.
](https://github.com/apache/airflow/pull/34715#issuecomment-1843640604)
I'm feeling lost on how to proceed and would appreciate some advice.

[Here's the link to the PR.](https://github.com/apache/airflow/pull/34715)

**Comment by potiuk** [comment, marked as answer]
Just raise your concern in the PR againg. It's normal to have people to respond with delays and miss things - you have exactly ONE pr to look at - but maintainers have ~200 opened + ~100 issues/disscussions a day opened and have to do deal with that. Which means that the fact you are waiting for something, migh tbe missed. But if you raise your concerns again, and remind, people will get reminded there. Usually what also help is periodic rebasing in your PR and making it "green" - even if is not in final state and might be changed, because often we assume that if PR is "red", it is still being worked on - the static checks and conflicts is a sign that something is not yet "ready for final review". 

Also - it's ok for people to have doubts and different opinions, even maintainers, you have to get used to it. It's a bit your job as an author to nudge them and pull their leg and if you see they disagree ask them to come to conclusions. 

So ... be persistent, remind gently (but remember that maintainers have their own lives and most of them respond in their free time, so the iteration might take time, sometimes long time.  Do your part - i.e. rebase, fix, add tests, make the PR green and respond to the comments raised. If you are unclear - write about it in the PR.
